src/gt4py/stencil_object.py

```diff
@@ -294,17 +296,17 @@ def _make_origin_dict(origin: Any) -> Dict[str, Index]:
             if origin is None:
                 return {}
             if isinstance(origin, collections.abc.Iterable):
-                return {"_all_": Index.from_value(origin)}
+                return {"_all_": cast(Tuple[int, ...], Index.from_value(origin))}
```

Comment:
This is just an idea not a requirement because it might involve a lot of work: it would be conceptually better to declare `NumericTuple` as a generic subclass of tuple (`Tuple[T, ...]`) and add the missing typing hints. Concrete subclasses could then be defined as concrete instances of `NumericTuple` (for example `class Index(NumericTuple[int]):`

Reply:
This sounds like a great idea. Let's try it in a separate PR.

Reply:
:+1: